search/typesenseGoandJS/search.js

```diff
@@ -0,0 +1,60 @@
+const Typesense = require('typesense')
+
+class TypeSenseResults{
+    constructor(results) {
+        this._rawresults= results;
+    }
+    length(){
+        this._length= this._rawresults["found"]
+        return this._length
+    }
+    results(){
+        let hits=(this._rawresults)["hits"];
+        let listofdocuments = []
+        for (let i =0; i<(hits).length;i++){
+            let doc = hits[i]["document"]
+            listofdocuments.push(doc)
+        }
+        this._listofresults=listofdocuments
+        return this._listofresults
+    }
+    resultsByType(){
+        let results = this.results()
```

Comment:
why not add if _this.resultsByType != null, so you don't have to recompute each time